### PR TITLE
[FIX] payment: initialize txContext before await

### PR DESCRIPTION
**Pull request by thle-odoo**
Steps to reproduce:
- install website_sale;
- install website_sale_picking;
- enable just "Pay in store when picking the product" provider;
- pusblish just "[On Site Pick] My Shop 1" shipping method;
- go to ecommerce and make the purchase flow.

Issue:
A traceback appears on the "/shop/payment" page.

Cause:
We use the `txContext` field in the `_setPaymentFlow` method before it is initialized.
Indeed during the `start` method, we have to wait for the end of the super before `txContext` is initialized.
Since the `start` method is asynchronous, this will not block the `_setPaymentFlow` method call which will use the uninitialized field.

opw-3257641

### CODE REVIEW

**Review by hubvd — APPROVED** [review]
Another way would be to declare `this.txContext` inside the `init` method, but  due to the way mixins works, it woudn't be possible.
So, this is the best fix we can do :+1:

File: addons/payment/static/src/js/payment_form_mixin.js [hunk 1]
```diff
@@ -12,14 +12,14 @@ odoo.define('payment.payment_form_mixin', require => {
          * @override
          */
         start: async function () {
+            this.txContext = {};
```
Comment (AntoineVDV) [inline addons/payment/static/src/js/payment_form_mixin.js:15]: To give a chance for the line to stay where it is.
```suggestion
            this.txContext = {};  // Synchronously initialize txContext before any await.
```

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
Hello, I could not reproduce the bug while following the steps in the commit description. What am I missing? Is this issue specific to website_sale_picking?

Also, the PR should probably target a lower version than 16.0.

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**